### PR TITLE
fix precision and recall calculations

### LINKED ISSUE #323: Bug: Precision and recall calculations
**Description:**
I discovered that the Precision and Recall have their logic reversed, the Precision metric is actually calculating Recall, and the Recall metric is calculating Precision.

**The Issue:**
In the Precision implementation, when counting false positives in binary classification:

```
if y_pred.get(i) == y_true.get(i) {
    if classes == 2 {
        if *y_true.get(i) == T::one() {
            tp += 1;
        }
    } else {
        tp += 1;
    }
} else if classes == 2 {
    if *y_true.get(i) == T::one() {
        fp += 1;
    }
}
```

This checks if y_true == 1 when predictions don't match. However, this is counting false negatives (when true=1 but pred=0), not false positives. False positives should be when we predict 1 but true label is 0.

Similarly, in the Recall implementation, when counting false negatives:
```

if y_pred.get(i) == y_true.get(i) {
    if classes == 2 {
        if *y_true.get(i) == T::one() {
            tp += 1;
        }
    } else {
        tp += 1;
    }
} else if classes == 2 {
    if *y_true.get(i) != T::one() {
        fne += 1;
    }
}
```

This checks if y_true == 0 when predictions don't match, which actually counts false positives (when true=0 but pred=1), not false negatives.

**Impact:**

- Precision scores are actually Recall scores
- Recall scores are actually Precision scores
- F-beta scores with beta ≠ 1 are incorrect

### PR DESCRIPTION
**Pull request by cmmartin**
Fixes #323

### Checklist
- [x] My branch is up-to-date with development branch.
- [x] Everything works and tested on latest stable Rust.
- [x] Coverage and Linting have been applied 

### Current behaviour
- **recall**
  - binary case calculates precision instead of recall due to incorrect false negative counting
  - multiclass case computes accuracy (total correct / total) instead of macro-averaged recall
- **precision**
  - binary case calculates recall instead of precision due to incorrect false positive counting
  - multiclass case computes accuracy instead of macro-averaged precision

### New expected behaviour
- **recall**
  - binary case correctly calculates recall
  - multiclass case computes macro-averaged recall instead of accuracy
  - handles zero-division edge case
- **precision**
  - binary case correctly calculates precision
  - multiclass case computes macro-averaged precision instead of accuracy
  - handles zero-division edge case

### CODE REVIEW

**Comment by Mec-iS** [comment]
thanks a lot. I will review this on monday

File: src/metrics/precision.rs [hunk 1]
```diff
@@ -114,7 +147,7 @@ mod tests {
         let y_pred: Vec<f64> = vec![0., 0., 1., 1., 1., 1.];
 
         let score3: f64 = Precision::new().get_score(&y_true, &y_pred);
-        assert!((score3 - 0.6666666666).abs() < 1e-8);
+        assert!((score3 - 0.5).abs() < 1e-8);
```
Comment (Mec-iS) [inline src/metrics/precision.rs:150]: it would be nice to take the opportunity and make these statements to use `approx` where possible

**Review by Mec-iS — APPROVED** [review]
💯 
Hope you will stick around!

**Comment by Mec-iS** [comment]
I will merge this and create a new release
